web/src/pages/HomePage.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Flex,
+  Grid,
+  GridItem,
+  Heading,
+  Image,
+  Link,
+  Stack,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+  useColorModeValue,
+  useMediaQuery,
+} from "@chakra-ui/react";
+import React from "react";
+import { useNavigate } from "react-router-dom";
+import { useRecoilValue } from "recoil";
+
+import DashboardControllerIamge from "@/assets/dashboard-controller-snapshot.svg";
+import GraphicalBackground2 from "@/assets/graphical-background-design-2.svg";
+import SingleStoreLogoDrak from "@/assets/singlestore-logo-dark.svg";
+import SinglestoreLogo from "@/assets/singlestore-logo-filled-sm.svg";
+import { DatabaseConfigForm } from "@/components/DatabaseConfigForm";
+import { useUpdateCityList } from "@/data/models/useUpdateCityList";
+import { connectionConfig } from "@/data/recoil";
+import { useConnectionState } from "@/view/hooks/hooks";
+
+export const HomePage: React.FC<{
+  redirectToHomePage: boolean;
+  setRedirectToHomaePage: (setStatus: boolean) => void;
+}> = ({ redirectToHomePage, setRedirectToHomaePage }) => {
+  const { connected } = useConnectionState();
+  const navigate = useNavigate();
+  const fontColor = useColorModeValue("#553ACF", "#CCC3F9");
+  const config = useRecoilValue(connectionConfig);
+  const { updateCityList } = useUpdateCityList(config);
+  const [isSmallScreen] = useMediaQuery("(max-width: 640px)");
+
+  React.useEffect(() => {
+    if (connected) {
+      setRedirectToHomaePage(false);
+    }
+  }, [connected, setRedirectToHomaePage]);
+
+  React.useEffect(() => {
+    if (!redirectToHomePage) {
+      updateCityList();
+      const urlSearchParams = new URLSearchParams(location.search.slice(1));
+      const queryObject = JSON.parse(
+        JSON.stringify(Object.fromEntries(urlSearchParams))
+      );
+      if (queryObject.redirect) {
+        navigate(queryObject.redirect);
```

Comment:
When using `navigate` you should specify that it should replace the current path, otherwise it will simply append it and can cause issues.

```suggestion
        navigate(queryObject.redirect, { replace: true });
```

Reply:
I didn't want to I want to append to existing base URL